File: frontend/src/companies/dto/requests/CompanyJoinRequestResponseDto.ts
import {CompanyResponsePublicDto} from "../CompanyResponsePublicDto.ts";
import {
  UserResponsePublicDto
} from "../../../authentication/dto/applicationUser/UserResponsePublicDto.ts";
import {RequestStatus} from "../RequestStatus.ts";

export interface CompanyJoinRequestResponseDto {
  readonly requestId: number;
  readonly company: CompanyResponsePublicDto;
  readonly user: UserResponsePublicDto;
  readonly status: RequestStatus;
}

File: backend/src/main/java/com/codecool/training_portal/dto/requests/CompanyJoinRequestResponseDto.java
package com.codecool.training_portal.dto.requests;

import com.codecool.training_portal.dto.company.CompanyResponsePublicDTO;
import com.codecool.training_portal.dto.user.UserResponsePublicDto;
import com.codecool.training_portal.model.request.RequestStatus;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;

public record CompanyJoinRequestResponseDto(
  @NotNull @Min(1) Long requestId,
  @NotNull CompanyResponsePublicDTO company,
  @NotNull @Valid UserResponsePublicDto user,
  @NotNull RequestStatus status) {
}
